internal/core/metadata/init.go

```diff
@@ -62,5 +58,12 @@ func BootstrapHandler(wg *sync.WaitGroup, ctx context.Context, startupTimer star
 		},
 		endpoint.Endpoint{RegistryClient: &registryClient})
 
+	// add dependencies to container
+	dic.Update(di.ServiceConstructorMap{
+		metadataContainer.ErrorHandlerName: func(get di.Get) interface{} {
+			return metadataContainer.ErrorHandlerName
+		},
```

Comment:
This map has a key with the service name and a ServiceConstructor-typed value that you're satisfying with a closure.  The name is correct.  The value implementation is not.  Your closure implementation is where the dependency is instantiated; it should provide an instance of errorconcept.Handler (it's currently providing a string).

While the code compiles and runs, it panics when the blackbox testing metadata importer is run:

```
2019/11/15 07:09:39 http: panic serving 127.0.0.1:57743: interface conversion: interface {} is string, not *errorconcept.Handler
```

Look at [this](https://github.com/edgexfoundry/edgex-go/blob/3fb26d3d713f167aaf69d4122db684b602a4c852/internal/system/agent/init.go#L53) for an example implementation.